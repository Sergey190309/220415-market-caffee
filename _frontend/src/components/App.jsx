import React from 'react'
import {Button} from 'semantic-ui-react'

const App = () => <Button content='App'/>


export default App