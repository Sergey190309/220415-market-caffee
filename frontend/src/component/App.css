.App {
  min-height: 100vh;
  /* background-color: red; */
  color: white;
  background-color: #282c34;
  position: relative;
}