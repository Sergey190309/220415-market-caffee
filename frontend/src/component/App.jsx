import React, { useState, useEffect, Fragment } from 'react'
import { Routes, Route, useNavigate } from 'react-router-dom'
import PropTypes from 'prop-types'
import { useDispatch } from 'react-redux'

import { setDeviceSize } from '../redux/slices'

// import NavBar from './navigation/NavBar'
// import Toggle from './navigation/Toggle'
import NavBarToggle from './navigation/NavBarToggle'
import Logo from './navigation/Logo'
import LanguageSwitcher from './general_items/language/LanguageSwitcher'

import AdminView from './page_views/admin/AdminView'
import LandingView from './page_views/landing/LandingView'
import PicturesView from './page_views/pictures/PicturesView'
import PriceListView from './page_views/priceList/PriceListView'
import UsersOnlyView from './page_views/usersOnly/UsersOnlyView'
import { GlobalStyle } from './styles/global.styled'

// import * as SB from './styles/buttons.styled'
// import { GlobalStyle, MainContainer, MainItem, GlobalDiv } from './styles/global.styled'
import NavBar from './navigation/NavBar'
// import NavItem from './navigation/NavItem'

export const switchNav = (setNavOpened, navOpened) => {
  setNavOpened(!navOpened)
}

const App = ({ setDeviceSize }) => {
  const [navOpened, setNavOpened] = useState(false)
  const [width, setWidth] = useState(window.innerWidth)
  const dispatch = useDispatch()
  // console.log('component>App setDeviceSize ->', setDeviceSize)

  const navigate = useNavigate()
  const toLanding = () => { navigate('/') }

  const setDinamicWidth = () => {
    setWidth(window.innerWidth)
  }
  useEffect(() => {
    window.addEventListener('resize', setDinamicWidth)
    dispatch(setDeviceSize(width))
    return () => window.removeEventListener('resize', setDinamicWidth)
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [width])


  const onClickToggleHandler = event => {
    // console.log('App>onClickToggleHandler, event ->', event.target)
    setNavOpened(true)
  }

  // console.log('App, rendering, navOpened ->', navOpened)

  return (
    <Fragment>
      <GlobalStyle />
      <Logo toLanding={toLanding} />
      <NavBarToggle switchNavBar={onClickToggleHandler} />
      <LanguageSwitcher />
      <NavBar
        visibility={navOpened}
        setVisibility={setNavOpened}
      />
      <Routes>
        <Route path='/' element={<LandingView />} />
        <Route path='/pricelist' exact element={<PriceListView />} />
        <Route path='/pictures' exact element={<PicturesView />} />
        <Route path='/private' exact element={<UsersOnlyView />} />
        <Route path='/admin' exact element={<AdminView />} />
      </Routes>
    </Fragment>
  )
}

App.defaultProps = {
  // techToken: '',
  setDeviceSize: setDeviceSize
}

App.propTypes = {
  setDeviceSize: PropTypes.func.isRequired
}

export default App