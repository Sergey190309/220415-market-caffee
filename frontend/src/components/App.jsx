import React, { Fragment, useState, useEffect } from 'react'
// Redux
// import { connect } from 'react-redux';
import PropTypes from 'prop-types'
import { useDispatch } from 'react-redux'

import Layout from './layout/Layout'

import { setDeviceSize } from '../redux/slices/device'
// import { setDeviceSize } from '../redux/actions';

export const App = ({ setDeviceSize }) => {
  const [width, setWidth] = useState(0)
  const dispatch = useDispatch()

  useEffect(() => {
    updateDimentions()
    window.addEventListener('resize', updateDimentions)
    return () => window.removeEventListener('resize', updateDimentions)
  }, [width])

  // useEffect(() => {
  //   getLngList(techToken);
  // }, [techToken]);

  const updateDimentions = () => {
    const _width = window.outerWidth
    setWidth(_width)
    dispatch(setDeviceSize(_width))
  }

  return (
    <Fragment>
      <Layout />
    </Fragment>
  )
}

App.defaultProps = {
  // techToken: '',
  setDeviceSize: setDeviceSize
}

App.propTypes = {
  setDeviceSize: PropTypes.func.isRequired
}

export default App
// export default connect(null, { setDeviceSize })(App);
