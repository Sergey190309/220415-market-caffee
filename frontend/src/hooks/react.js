import {useState, useEffect, useRef, useCallback } from 'react'

export const useAppState = useState
export const useAppEffect = useEffect
export const useAppRef = useRef
export const useAppCallback = useCallback
