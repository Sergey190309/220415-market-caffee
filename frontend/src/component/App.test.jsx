import React from 'react'

import TestRenderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'

import App from './App'

describe('App', () => {
  test('Snapshot', () => {
    // const something = TestRenderer
    const element = TestRenderer.create(<App />)

    expect(element).toMatchSnapshot()
    // await waitFor(() => {

    // })
    // console.log('App ->', container)
    // screen.debug()
    // const tree = component.toJSON()
    // expect(tree).toMatchSnapshot()
  })
  test('Structure', () => {
    render(<App />)
    const button = screen.getByRole('button')
    expect(button).toBeInTheDocument()
    expect(button).toHaveTextContent("That's button!")
    // screen.debug()
  })
})
