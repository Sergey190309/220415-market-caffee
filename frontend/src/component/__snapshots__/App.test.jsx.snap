// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot 1`] = `
Array [
  <div
    className="sc-gsnTZi kCDofj"
  >
    <div
      className="sc-dkzDqf irZllQ"
    >
      it works!
    </div>
    <div
      className="sc-dkzDqf irZllQ"
    >
      header
    </div>
    <div
      className="sc-dkzDqf irZllQ"
    >
      hBlock
    </div>
    <div
      className="sc-dkzDqf irZllQ"
    >
      vBlock
    </div>
    <div
      className="sc-dkzDqf irZllQ"
    >
      Footer
    </div>
  </div>,
  <button
    className="sc-bczRLJ etCJzT"
  >
    That's button!
  </button>,
]
`;
